cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -236,6 +248,122 @@ func canInstallAndRunAtOnce(mst *initramfsMountsState) (bool, error) {
 	return kernelHasFdeSetup, nil
 }
 
+func readSnapInfo(sysSnaps map[snap.Type]*seed.Snap, snapType snap.Type) (*snap.Info, error) {
+	seedSnap := sysSnaps[snapType]
+	mountPoint := filepath.Join(boot.InitramfsRunMntDir, snapTypeToMountDir[snapType])
+	return snap.ReadInfoFromMountPoint(seedSnap.SnapName(), mountPoint, seedSnap.Path, seedSnap.SideInfo)
+}
+
+func runFDESetupHook(req *fde.SetupRequest) ([]byte, error) {
```

Comment:
we want a TODO about making this into a systemd-run using helper from kernel/fde

Reply:
Done.